docker/test-server/setup.sh

```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
+
+set -ex
+
+apk add curl libpq postgresql-dev --virtual .build-deps coreutils gcc libc-dev libffi-dev make
+
+ETESYNC_VERSION=$1
+if test -z "${ETESYNC_VERSION}"; then
+    echo "set --build-arg ETESYNC_VERSION"
+    exit 1
+fi
+
+mkdir -p /app /data/static /data/media
+
+curl -v -ssL --fail https://github.com/etesync/server/archive/refs/tags/v${ETESYNC_VERSION}.tar.gz | tar -C /app --strip-components=1 -zxf -
```

Comment:
I would prefer it if it used the local sources rather than the released version, because:
1. I trust their integrity better.
2. It'll enable us to push a "latest" image that tracks git.

Reply:
Sounds good.  At least for users of the clients -- and I think for the clients as well -- they should use a released version, though, to avoid accidentally requiring unreleased features.

Reply:
I think they should use latest, because if we add something to a client we usually also add it to the server in tandem. So having code in a client that depends on latest server is actually quite common.

Reply:
Ok, makes sense.  I've set this up so it builds HEAD, so it should be easy to produce both `latest` and versioned tags.

Reply:
I just updated it to take a rev argument, so building a tag should be pretty easy, too.  It builds from a `git archive` to avoid including any cruft from a local working copy that might cause variations between builds.

Reply:
Yeah, I'm not a fan as it's non-standard.